crates/ruff/src/rules/pycodestyle/rules/type_comparison.rs

```diff
@@ -44,55 +46,56 @@ pub(crate) fn type_comparison(
     ops: &[CmpOp],
     comparators: &[Expr],
 ) {
-    for (op, right) in izip!(ops, comparators) {
+    for (op, right) in ops.iter().zip(comparators) {
         if !matches!(op, CmpOp::Is | CmpOp::IsNot | CmpOp::Eq | CmpOp::NotEq) {
             continue;
         }
         match right {
             Expr::Call(ast::ExprCall {
-                func,
-                arguments: Arguments { args, .. },
-                ..
+                func, arguments, ..
             }) => {
-                if let Expr::Name(ast::ExprName { id, .. }) = func.as_ref() {
-                    // Ex) `type(False)`
-                    if id == "type" && checker.semantic().is_builtin("type") {
-                        if let Some(arg) = args.first() {
-                            // Allow comparison for types which are not obvious.
-                            if !matches!(
-                                arg,
-                                Expr::Name(_)
-                                    | Expr::Constant(ast::ExprConstant {
-                                        value: Constant::None,
-                                        kind: None,
-                                        range: _
-                                    })
-                            ) {
-                                checker
-                                    .diagnostics
-                                    .push(Diagnostic::new(TypeComparison, expr.range()));
-                            }
-                        }
-                    }
-                }
-            }
-            Expr::Attribute(ast::ExprAttribute { value, .. }) => {
-                if let Expr::Name(ast::ExprName { id, .. }) = value.as_ref() {
-                    // Ex) `types.NoneType`
-                    if id == "types"
-                        && checker
-                            .semantic()
-                            .resolve_call_path(value)
-                            .is_some_and(|call_path| {
-                                call_path.first().is_some_and(|module| *module == "types")
-                            })
+                // Ex) `type(obj) is type(1)`
+                let Expr::Name(ast::ExprName { id, .. }) = func.as_ref() else {
+                    continue;
+                };
+
+                if id == "type" && checker.semantic().is_builtin("type") {
+                    // Allow comparison for types which are not obvious.
+                    if arguments
+                        .args
+                        .first()
+                        .is_some_and(|arg| !arg.is_name_expr() && !is_const_none(arg))
```

Comment:
Nit: I don't know how expensive `is_builtin` is, but we could potentially move this check out (or avoid nested ifs ;))